config/config_test.go

```diff
@@ -228,6 +228,59 @@ func TestDeprecateLogMessage(t *testing.T) {
 	})
 }
 
+func TestBadConfigKeys(t *testing.T) {
```

Comment:
If you decide to pull the validation logic out from `config.go`, the testing can also go to their dedicated file too